sql: add switch for check constraint (#13998)
pingcap/docs: constraints.md

## CHECK

> **Note:**
>

> > The 'CHECK' constraint feature is disabled by default. To enable it, you need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.

Reviewer: ```suggestion
> The `CHECK` constraint feature is disabled by default. To enable it, you need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.
```

## CHECK

> **Note:**
>

> > 'CHECK' constraint function is off by default, [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-New in v7.2.0) needs to be enabled before it takes effect.

Reviewer 2: ```suggestion
> The 'CHECK' constraint feature is disabled by default. To enable it, You need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.
```

> > The 'CHECK' constraint feature is disabled by default. To enable it, You need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.

Reviewer 2: ```suggestion
> The 'CHECK' constraint feature is disabled by default. To enable it, you need to set the [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-new-in-v720) variable to `ON`.
```

> 

Reviewer 2: ```suggestion
```

ALTER TABLE t ALTER CONSTRAINT c1 NOT ENFORCED;
```

### MySQL compatibility

> - It is not supported to add a `CHECK` constraint while adding a column (for example, `ALTER TABLE t ADD COLUMN a CHECK(a > 0)`). In this case, only the column will be added successfully, and TiDB will ignore the `CHECK` constraint without reporting any error.
> - It is not supported to use `ALTER TABLE t CHANGE a b int CHECK(b > 0)` to add a `CHECK` constraint. When this statement is executed, TiDB will report an error.

Reviewer: ```suggestion
- It is not supported to add a `CHECK` constraint while adding a column (for example, `ALTER TABLE t ADD COLUMN a CHECK(a > 0)`). In this case, only the column is added successfully, and TiDB ignores the `CHECK` constraint without reporting any error.
- It is not supported to use `ALTER TABLE t CHANGE a b int CHECK(b > 0)` to add a `CHECK` constraint. When this statement is executed, TiDB reports an error.
```

ALTER TABLE t ALTER CONSTRAINT c1 NOT ENFORCED;
```

> ### MySQL Compatibility

Reviewer 2: ```suggestion
### MySQL compatibility
```

> - It is not supported to add a `CHECK` constraint while adding a column in (for example, `ALTER TABLE t ADD COLUMN a CHECK(a > 0)`), otherwise only the column will be added successfully, and TiDB will ignore the `CHECK` constraint and will not report an error.

Reviewer 2: ```suggestion
- It is not supported to add a `CHECK` constraint while adding a column (for example, `ALTER TABLE t ADD COLUMN a CHECK(a > 0)`). In this case, only the column will be added successfully, and TiDB will ignore the `CHECK` constraint without reporting any error.
```

> - It is not supported to use `ALTER TABLE t CHANGE a b int CHECK(b > 0)` to add a `CHECK` constraint, and TiDB will report an error when using this statement.

Reviewer 2: ```suggestion
- It is not supported to use `ALTER TABLE t CHANGE a b int CHECK(b > 0)` to add a `CHECK` constraint. When this statement is executed, TiDB will report an error.
```
